pkg/notes/notes.go

```diff
@@ -679,6 +679,42 @@ func HasString(a []string, x string) bool {
 	return false
 }
 
+func getPRNumberFromCommit(client *github.Client, logger log.Logger, commit *github.RepositoryCommit, opts ...GithubApiOption) (int, error) {
+	num, err := getPRNumberFromCommitMessage(*commit.Commit.Message)
+	if err != nil {
+		level.Debug(logger).Log(
+			"err", err,
+			"msg", "error getting the pr number from commit message",
+			"sha", commit.GetSHA(),
+		)
+		return getPRNumberFromCommitSHA(client, *commit.SHA, opts...)
```

Comment:
What happens if a PR contains multiple commits and re use a rebase-based merge approach? The commits would occur twice, right? But I think they would be de-duped afterwards because of:
https://github.com/kubernetes/release/blob/3a09baa0f5b12b2596ce1501af886a60ffc3b494/pkg/notes/notes.go#L218-L221
Do we want to pre-check if the PR has already been added?